### ISSUE TITLE
single<T> { existingInstance } not recognized by compiler plugin

### ISSUE THREAD

**Issue by ymloh01** [InsertKoinIO/koin-compiler-plugin#36]
I am using Koin with the compiler plugin, and I’m trying to register an already-created `Vertx` instance into a module.

```kotlin
val vertx: Vertx = Vertx.vertx()

fun provideVertx() = vertx
```

### Module definition

```kotlin
val myModule = module {

    // ❌ Not detected by compiler plugin
    single<Vertx> { vertx }

    // ❌ Not detected by compiler plugin
    single<Vertx> { provideVertx() }

    // ❌ Not detected by compiler plugin
    single<Vertx> { create { -> vertx } }

    // ✅ Only this works
    single<Vertx> { create(::provideVertx) }
}
```

### Observed behavior

All variants except `create(::provideVertx)` fail at compile time with:

```
[ERROR] [Koin] Missing dependency: io.vertx.core.Vertx
```

### Expected behavior

All of the following should be equivalent and accepted by the compiler plugin:

```kotlin
single<Vertx> { vertx }
single<Vertx> { provideVertx() }
```

### Questions

1. Is `create(::namedFunction)` the intended/required API for registering existing instances, or is this a limitation of the compiler plugin’s static analysis?

2. Are there any plans to support direct instance bindings like:

   ```kotlin
   single<T> { existingInstance }
   ```

   without requiring `create(...)` or a function reference?

**Comment by abishai** [comment]
I have the same issue, but with ObjectMapper() I try to instantiate. Even more, provider function triggers UnsafeDSL check. So, I disabled CompilerSafety entirely. I hope, this is fixable, as I really want dependency graph analysis in compile time.

Annotations based approach work though.

**Comment by abishai** [comment]
Annotations has some issues as well, for example it seems I can't pass ktor environment to Koin, compiler plugin doesn't see it and required CompileSafety = false to initialize.
```
fun Application.configureDI() {
  install(Koin) {
    slf4jLogger()

    withConfiguration<KoinUserApplication> {
      modules(module { single { this@configureDI.environment } })
    }
  }
```

**Comment by Faltenreich** [comment]
+1 and for us this also affects the option with a function reference with Koin 4.2.2 and Koin Compiler Plugin 1.0.1.

We are trying to inject a CoroutineDispatcher and receive the following error:

```
[Koin][KOIN-D001] Missing dependency: kotlinx.coroutines.CoroutineDispatcher
```

None of the following works:

```
single<CoroutineDispatcher> { Dispatchers.IO }
single<CoroutineDispatcher>(Dispatchers::IO)
single<CoroutineDispatcher> { create { Dispatchers.IO } }
single<CoroutineDispatcher> { create(Dispatchers::IO) }
```

This means we are currently blocked from using compile-time safety in tandem with the Classic DSL. Is there another way to inject external dependencies?

Thanks in advance and keep up the good work!

**Comment by Faltenreich** [comment]
Just as we have found a workaround for this issue, it got closed. What a coincidence!

This also works with 1.0.1, with or without function reference:

```
single<CoroutineDispatcher> { create(::getDispatcher) }

fun getDispatcher(): CoroutineDispatcher = Dispatchers.IO
```

**Comment by ymloh01** [comment]
Is the fix for this has been officially released? I noticed it's tagged for version 1.0.2, but that specific version does not seem to be available on Maven Central

**Comment by arnaudgiuliani** [comment]
It's publishing 👍
